### PR TITLE
RPRBLND-2016: Improve Material X editor workflow

### PR DESCRIPTION
**Pull request by VascoPi**
### PURPOSE
Need to fix material X Editor workflow, because now MaterialX selection doesn't work as expected.

### EFFECT OF CHANGE
Improved MaterialX nodetree editor behavior when selecting MaterialX through Material Properties.

### TECHNICAL STEPS
Added update_material_iu function.
Added turn on/off for bpy.types.NODE_HT_header updates.
Added condition for nodetree pin option.

### NOTES FOR REVIEWERS
There's a workaround for Blender 3.0 because of changing logic. For Blender 3.0 implemented only the pin option.

### CODE REVIEW

File: src/hdusd/utils/mx.py
Comment (bnagirniak) [inline src/hdusd/utils/mx.py:295]: Move this function to ui/material.py as more related.

**Review by bnagirniak — COMMENTED** [review]
Tested - works ok, suggested code improvement.

**Review by bnagirniak — CHANGES_REQUESTED** [review]
Error found in console:
```
Traceback (most recent call last):
  File "D:\blender\blender-2.93.0-f240b5e5f712-windows64\2.93\scripts\modules\bpy_types.py", line 807, in draw_ls
    func(self, context)
  File "D:\amd-gpuopen\BlenderUSDHydraAddon\src\hdusd\utils\mx.py", line 302, in update_material_iu
    if space.node_tree != mat.node_tree:
AttributeError: 'NoneType' object has no attribute 'node_tree'
```

**Review by DagerD — CHANGES_REQUESTED** [review]
If we have MatX attached and then try to switch from shader nodetree to MatX nodetree, MatX will be detached and swtich won't happen.

**Comment by VascoPi** [comment]
Please don't merge till new usability isn't approved by Q&A .

**Comment by bnagirniak** [comment]
Let's wait for @ARozghon to test

**Review by bnagirniak — APPROVED** [review]
Tested - works ok

File: src/hdusd/ui/material.py
Comment (DagerD) [inline src/hdusd/ui/material.py:754]: typo "update_material_ui"

**Review by DagerD — COMMENTED** [review]
commented in code

**Review by DagerD — APPROVED** [review]

**Comment by bsavery** [comment]
@ARozghon  ?

**Comment by ARozghon** [comment]
Looked at the latest autotests and we have an issue with coatColor.png test case with RDNA and RDNA2 GPUs https://cis.nas.luxoft.com/BlenderUSDHydraPluginAuto/PR-177/19/Test_Report_RPR/AMD_RX6800-Windows-regression.json~/Results/BlenderUSDHydra/MaterialX_Standart_Surface/report.html?searchText=USD_BL_MTX_SS_058
https://cis.nas.luxoft.com/BlenderUSDHydraPluginAuto/PR-177/19/Test_Report_RPR/AMD_RX5700XT-Windows-regression.json~/Results/BlenderUSDHydra/MaterialX_Standart_Surface/report.html?searchText=USD_BL_MTX_SS_058
but not reproduce manually with 6800 XT and driver 22.1.2
please, doublecheck before the merge

**Comment by ARozghon** [comment]
manual tests on 6800 XT complete without any issues

**Review by ARozghon — APPROVED** [review]
manually working fine, need to update autotests

**Pull request merged**
